EFCore.NamingConventions/Internal/NameRewritingConvention.cs

```diff
@@ -61,7 +127,7 @@ public void ProcessForeignKeyOwnershipChanged(IConventionForeignKeyBuilder relat
                     .Except(ownedEntityType.FindPrimaryKey().Properties)
                     .Where(p => p.Builder.CanSetColumnName(null)))
                 {
-                    var columnName = property.GetColumnName();
+                    var columnName = property.GetColumnBaseName();
                     var prefix = _namingNameRewriter.RewriteName(ownedEntityType.ShortName());
                     if (!columnName.StartsWith(prefix, StringComparison.OrdinalIgnoreCase))
```

Comment:
Might make sense to also check for the non-rewritten prefix

Reply:
Thanks, added that.